packages/@uppy/audio/src/index.js

```diff
@@ -0,0 +1,403 @@
+const { h } = require('preact')
+const { UIPlugin } = require('@uppy/core')
+const getFileTypeExtension = require('@uppy/utils/lib/getFileTypeExtension')
+const supportsMediaRecorder = require('./supportsMediaRecorder')
+const RecordingScreen = require('./RecordingScreen')
+const PermissionsScreen = require('./PermissionsScreen')
+const locale = require('./locale.js')
+const packageJsonVersion = require('../package.json').version
+
+/**
+ * Setup getUserMedia, with polyfill for older browsers
+ * Adapted from: https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia
+ */
+function getMediaDevices () {
+  // eslint-disable-next-line compat/compat
+  if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
+    // eslint-disable-next-line compat/compat
+    return navigator.mediaDevices
+  }
+
+  const getUserMedia = navigator.mozGetUserMedia || navigator.webkitGetUserMedia
+  if (!getUserMedia) {
+    return null
+  }
+
+  return {
+    getUserMedia (opts) {
+      return new Promise((resolve, reject) => {
+        getUserMedia.call(navigator, opts, resolve, reject)
+      })
+    },
+  }
+}
+/**
+ * Audio
+ */
+module.exports = class Audio extends UIPlugin {
+  static VERSION = packageJsonVersion
+
+  constructor (uppy, opts) {
+    super(uppy, opts)
+    this.mediaDevices = getMediaDevices()
+    this.supportsUserMedia = !!this.mediaDevices
+    // eslint-disable-next-line no-restricted-globals
+    this.protocol = location.protocol.match(/https/i) ? 'https' : 'http'
```

Comment:
Can we get rid of the eslint warning here:
```suggestion
    this.protocol = globalThis.location.protocol.match(/https/i) ? 'https' : 'http'
```